src/Tribe/Views/V2/Views/Day_View.php

```diff
@@ -146,4 +146,58 @@ protected function build_url_for_date( $url_date, $canonical, array $passthru_va
 
 		return $url;
 	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function setup_template_vars() {
+
+		$template_vars = parent::setup_template_vars();
+		$sorted_events = $this->sort_events( $template_vars['events'] );
+
+		$template_vars['events'] = $sorted_events;
+
+		return $template_vars;
+	}
+
+	/**
+	 * Add timeslot and sort events for the day view.
+	 *
+	 * Iterate over the day events to add timeslots and sort them.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $events  An array of events.
+	 *
+	 * @return array The sorted and modified array.
+	 */
+	protected function sort_events( $events ) {
+
+		foreach ( $events as $event ) {
+			if ( ! empty( $event->all_day ) ) {
+				$event->timeslot = esc_html__( 'All Day', 'the-events-calendar' );
+			} elseif ( ! empty( $event->multiday ) ) {
+				$event->timeslot = esc_html__( 'Ongoing', 'the-events-calendar' );
+			}
+		}
+
+		// Make sure All Day and ongoing events come first.
+		$all_day = [];
+		$ongoing = [];
+		$hourly  = [];
+		foreach ( $events as $i => $event ) {
```

Comment:
We should avoid a loop here, maybe using `usort()` based on a `timeslot` comparison